src/main/scala/com/gu/contentapi/utils/PredicateUtils.scala

```diff
@@ -0,0 +1,20 @@
+package com.gu.contentapi.utils
+
+object PredicateUtils {
+
+  implicit class RichPredicate[A](val f: Function1[A, Boolean]) extends AnyVal {
+    def apply(v: A) = f(v)
+
+    def &&(g: Function1[A, Boolean]): Function1[A, Boolean] = {
```

Comment:
Yes. The reason is that we are extending `Function1[A, Boolean]` with new methods ([extension methods](http://docs.scala-lang.org/overviews/core/value-classes.html#extension-methods)). As you know `=>` is just syntaxic sugar that is represented internally by `FunctionN` types.

So in general I will agree with you that `FunctionN` methods should not be visible, but in this particular case we are extending the plumbing.